src/nunit.analyzers/ClassicModelAssertUsage/ClassicModelAssertUsageAnalyzer.cs

```diff
@@ -45,10 +45,9 @@ private static void AnalyzeInvocation(SyntaxNodeAnalysisContext context)
             if (methodNode != null && !methodNode.ContainsDiagnostics)
             {
                 var invocationNode = (InvocationExpressionSyntax)context.Node;
-                var invocationSymbol = context.SemanticModel.GetSymbolInfo(
-                    invocationNode.Expression).Symbol as IMethodSymbol;
 
-                if (invocationSymbol != null && invocationSymbol.ContainingType.IsAssert())
+                if (context.SemanticModel.GetSymbolInfo(
+                        invocationNode.Expression).Symbol is IMethodSymbol invocationSymbol && invocationSymbol.ContainingType.IsAssert())
```

Comment:
I found the original code easier to read. I'm unsure if placing `if (context.SemanticModel.GetSymbolInfo(invocationNode.Expression).Symbol is IMethodSymbol invocationSymbol &&` on one line and the rest on the next line would make it easier. But still I find that several member accesses, two method calls, pattern matching, and a boolean operator to be too much "work" inside an `if`-statement.